docs/ldml/tr35-personNames.md

```diff
@@ -41,26 +41,924 @@ The LDML specification is divided into the following parts:
+
+There may be more modifiers in the future. 
+
+Examples: 
+
+1. For the initial of the surname “**_de Souza_**”, in a language that treats the “de” as a tussenvoegsel, the PersonName object can automatically recast `{surname-initial}` to:<br>`{surname-prefix-initial}{surname-core-initial-allCaps} `to get “dS” instead of “d”.
+2. If the locale expects a surname prefix to to be sorted after a surname, then both `{surname-core} `then `{surname-prefix}` would be used as in<br>`{surname-core}, {given} {given2} {surname-prefix}`
+
+## 6 <a name="6-formatting-process" href="#6-formatting-process">Formatting Process</a>
+
+The patterns are in personName elements, which are themselves in a personNames container element. The following describes how these patterns are chosen. If the name locale is different than the formatting locale, then additional processing needs to take place: see [Handling foreign names](#6-8-handling-foreign-names).
+
+The details of the XML structure behind the data referenced here are in [XML Structure](#2-xml-structure).
+
+### 6.1 <a name="6-1-derive-the-name-locale" href="#6-1-derive-the-name-locale">Derive the name locale</a>
+
+If there is no name locale available via the PersonName data interface, then an effective name locale is created based on the characters in the name and the formatting locale. 
+
+* Get the predominant script of the given and surname, and find the likely language as specified by the likely subtags. Use the locale identifier constructed from the likely language and the script as the name locale.
+
+### 6.2 <a name="6-2-derive-the-formatting-locale" href="#6-2-derive-the-formatting-locale">Derive the formatting locale</a>
+
+If the name locale is different from the formatting locale, and the script of the name is different from the script of the formatting locale, then change the formatting locale to the name locale.
+
+### 6.3 <a name="6-3-derive-the-name-order" href="#6-3-derive-the-name-order">Derive the name order</a>
+
+The name order data provides a mapping from the name locale to an order parameter.`  `
+
+A PersonName object’s fields are used to derive an order, as follows:
+
+1. If the PersonName object to be formatted has a `preferredOrder` field, then return that field’s value
+2. Otherwise, if the PersonName object has a `nameLocale` field, and there is a `nameOrderLocales` element:
+    1. Map the `nameLocale` field using the likelySubtags data, eg `zh` ⇒ `zh_Hans_CN`
+    2. Traverse the `nameOrderLocales` elements, and within each order attribute value to see if there is a match. 
```

Comment:
I think we're missing a couple steps here too.  We expand the name locale to include all the optional parts and search both `nameOrderLocales` list for the expanded name locale.  If we don't find it, we lop a field off the end and try again, and if we lop off all the fields, we search the elements for `und`.  The lopping-off part is missing.

But there are two wrinkles to that, I think:
1. Do we really need the script code here?  I'm a little worried it might interact uncomfortably with the script-based selection of name and formatting locales.
2. The truncation-fallback logic I outline above (and that I think both Mark and I implemented in our implementations) doesn't allow for `und` to be used as a wildcard language (so you can use `und_CN` to mean "any language, but in China".  I don't know if we're still mentioning this usage in this doc (I don't remember seeing it this far), but if we want to support that, we need a different algorithm.

Reply:
I suggest:

Replace
> 2. Otherwise, if the PersonName object has a `nameLocale` field, and there is a `nameOrderLocales` element:
>     1. Map the `nameLocale` field using the likelySubtags data, eg `zh` ⇒ `zh_Hans_CN`
>     2. Traverse the `nameOrderLocales` elements, and within each order attribute value to see if there is a match. 
By 

2. Otherwise use the nameOrderLocales elements to find the most best match for the full name locale, as follows.
  * For each locale L in the parent locale lookup chain* for the full name locale, do the following
    b. If there is a precise match among the givenFirst nameOrderLocales for L, then set the nameOrder to givenFirst, and stop.
    d. Otherwise if there is a precise match among the surnameFirst nameOrderLocales for the L, then set the nameOrder to givenFirst, and stop.
    e. Otherwise repeat (c) and (d) with "und" substituted for the language subtag in L.

* For example, here is a parent locale lookup chain: de_Latn_de ⇒ de_Latn ⇒ de_de ⇒ de ⇒ und

This process will always terminate, because there is always a `und` value in one of the two nameOrderLocales elements.

Reply:
>b. Otherwise if there is a precise match among the surnameFirst nameOrderLocales for the L, then set the nameOrder to givenFirst, and stop.

I think you mean "set the nameOrder to surnameFirst" here...

>c. Otherwise repeat (a) and (b) with "und" substituted for the language subtag in L.

That'll never match "und_XX" because the parent chain the first time through step a includes "und".  You might be able to get around this just by taking "und" out of your example parent chain (although you probably want to include some verbiage highlighting this case a little better).

Other than those things, this looks good to me.

Reply:
for #1, agreed.

You won't hit 'und' until the very end of the chain. The c step is on each iteration. That is:
```
for (L in chain) {
  for (L2 in L, L') {
    if there is a precise match for L2 in givenFirst locales, return givenFirst
    else if there is a precise match for L2 in surnameFirst locales, return surnameFirst
  }
}
```
(Where L' is obtained from L by replacing the language subtag by 'und'.)

Reply:
I agree that my first formulation was confusing. Suggestions for wording?

Reply:
Okay, yeah, I totally misread it.  I think you can clarify this by including an example at the end, maybe instead of the current parent-chain example, maybe after it:

>In other words, you'll check the `givenFirst` and `surnameFirst` resources for the following locales, in this order:
>
>de_Latin_DE, und_Latn_DE, de_Latn, und_Latn, de_DE, und_DE, de, und

Doing it this way biases the algorithm in favor of matching the country code (that is, `und_CN` will beat `zh` if you're searching for `zh_CN`), which I'm a little leery of, but that probably doesn't matter in practice because it wouldn't really make sense for somebody to have `zh` and `und_CN` (or whatever pair) have different name orders.